e2e_tests/test_browse.py

```diff
@@ -10,25 +10,318 @@ def test_has_title(authenticated_page: Page):
     )
 
 
-def test_sort_dropdown(authenticated_page: Page):
-    authenticated_page.goto("/browse")
-    authenticated_page.select_option("#sort", "body-a")
-    authenticated_page.select_option("#sort", "body-b")
-    authenticated_page.select_option("#sort", "series-a")
-    authenticated_page.select_option("#sort", "series-b")
-    authenticated_page.select_option("#sort", "date-first")
-    authenticated_page.select_option("#sort", "date-last")
+class TestBrowse:
+    def test_browse_sort_dropdown(self, authenticated_page: Page):
+        authenticated_page.goto("/browse")
+        authenticated_page.select_option("#sort", "transferring_body-asc")
+        authenticated_page.select_option("#sort", "transferring_body-desc")
+        authenticated_page.select_option("#sort", "series-asc")
+        authenticated_page.select_option("#sort", "series-desc")
+        authenticated_page.select_option(
+            "#sort", "last_record_transferred-desc"
+        )
+        authenticated_page.select_option("#sort", "last_record_transferred-asc")
 
+    def test_browse_filter_functionality_with_query_string_parameters(
+        self, authenticated_page: Page
+    ):
+        authenticated_page.goto(
+            "/browse?transferring_body_filter=all&series_filter=&date_from_day"
+            "01=&date_from_month=07&date_from_year=2023&date_to_day=31&date_to_month=07&date_to_year=2023"
+        )
 
-def test_filter_functionality(authenticated_page: Page):
-    authenticated_page.goto("/browse")
-    authenticated_page.select_option(
-        ".govuk-select__filters-form-transferring-body-select",
-        "Arts Council England",
-    )
+        table = authenticated_page.locator("#tbl_result").first
+        headers = table.locator("thead th").all_text_contents()
+        rows = table.locator("tbody tr")
+        cols = 0
+        if rows.count() > 0:
+            cols = rows.first.locator("td")
 
-    records = authenticated_page.query_selector_all(
-        ".govuk-summary-list__value a"
-    )
-    for record in records:
-        assert record
+        assert headers == [
+            "Transferring body",
+            "Series",
+            "Last record transferred",
+            "Records held",
+            "Consignments within series",
+        ]
+
+        assert rows.count() == 1
+
+        if cols.count() > 0:
```

Comment:
we should not have conditional statements in tests especially around assertions as this will mean we cant trust 2 runs of the test behave the same.

Reply:
removed condition as requested